api/proto/config/fw.proto

```diff
@@ -8,29 +8,42 @@ option java_package = "com.zededa.cloud.uservice.proto";
 
 
 message ACEMatch {
-	// FIXME: We should convert this to enum
-	string type = 1;
-	string value = 2;
+  // FIXME: We should convert this to enum
+  string type = 1;
+  string value = 2;
 }
 
 message ACEAction {
-	bool drop = 1;
+  bool drop = 1;
 
-	// limit action, and its associated parameter
-	bool limit = 2;
-	uint32 limitrate = 3;
-	string limitunit = 4;
-	uint32 limitburst = 5;
+  // limit action, and its associated parameter
+  bool limit = 2;
+  uint32 limitrate = 3;
+  string limitunit = 4;
+  uint32 limitburst = 5;
 
-	// port map action, and its assoicated paramtere
-	bool portmap = 6;
-	uint32 appPort = 7;
+  // port map action, and its assoicated paramtere
+  bool portmap = 6;
+  uint32 appPort = 7;
 }
 
-message ACE {
-	repeated ACEMatch matches = 1;
+enum ACEDirection {
+  BOTH = 0;
+  INGRESS = 1;
+  EGRESS = 2;
+}
 
-	// Expect only single action...repeated here is
-	// for future work.
-	repeated ACEAction actions = 2;
+message ACE {
+  // multiple matches here is for various fields of 6 tuples
+  //  for example
+  //     1) host=www.example.com & port=http
+  //     2) ip=8.8.8.8 & port=53 & proto=UDP
+  repeated ACEMatch matches = 1;
+
+  // Expect only single action...repeated here is
+  // for future work.
+  repeated ACEAction actions = 2;
+  string name = 3; // User visible name of the ACL
+  string id = 4;   // identifier
```

Comment:
string here and int32 in flowlog.
I think we want an int32 so we can feed it into the kernel which supports integer conntrack marks.

Reply:
fixed.